tracer/build/_build/Build.VariableGenerations.cs

```diff
@@ -0,0 +1,321 @@
+using System.Collections.Generic;
+using System.Linq;
+using Newtonsoft.Json;
+using Nuke.Common;
+using Nuke.Common.CI.AzurePipelines;
+using Nuke.Common.Tools.Git;
+using NukeExtensions;
+
+partial class Build : NukeBuild
+{
+    Target GenerateVariables
+        => _ =>
+        {
+            return _
+                  .Unlisted()
+                  .Executes(() =>
+                   {
+                       GenerateConditionVariables();
+
+                       GenerateIntegrationTestsWindowsMatrices();
+                       GenerateIntegrationTestsLinuxMatrix();
+                       GenerateExplorationTestMatrices();
+                       GenerateSmokeTestsMatrices();
+                   });
+
+            void GenerateConditionVariables()
+            {
+                GenerateConditionVariableBasedOnGitChange("isDebuggerChanged", new[] { "tracer/src/Datadog.Trace.ClrProfiler.Native/Debugger" });
+                GenerateConditionVariableBasedOnGitChange("isProfilerChanged", new[] { "profiler/src" });
+
+                void GenerateConditionVariableBasedOnGitChange(string variableName, string[] filters)
+                {
+                    var masterCommit = GitTasks.Git("merge-base origin/master HEAD").First().Text;
+                    var changedFiles =
+                            GitTasks
+                               .Git($"diff --name-only \"{masterCommit}\"")
+                               .Select(output => output.Text)
+                               .ToArray()
+                        ;
+
+                    var isChanged = filters.Any(filter => changedFiles.Any(s => s.Contains(filter)));
+
+                    Logger.Info($"{variableName} - {isChanged}");
+
+                    var variableValue = isChanged.ToString();
+                    EnvironmentInfo.SetVariable(variableName, variableValue);
+                    AzurePipelines.Instance.SetVariable(variableName, variableValue);
+                }
+            }
+
+            void GenerateIntegrationTestsWindowsMatrices()
+            {
+                var targetFrameworks = TargetFramework.GetFrameworks(except: new[] { TargetFramework.NETSTANDARD2_0 });
+
+                GenerateIntegrationTestsWindowsMatrix(targetFrameworks);
+                GenerateIntegrationTestsWindowsIISMatrix(targetFrameworks);
+            }
+
+            void GenerateIntegrationTestsWindowsMatrix(TargetFramework[] targetFrameworks)
+            {
+                var targetPlatforms = new[] { "x86", "x64" };
+                var matrix = new Dictionary<string, object>();
+
+                foreach (var framework in targetFrameworks)
+                {
+                    foreach (var targetPlatform in targetPlatforms)
+                    {
+                        matrix.Add($"{targetPlatform}_{framework}", new { framework, targetPlatform });
```

Comment:
Thinking out loud: I wonder if we should explicitly set these, as a rename here would break the generated json, which could be tricky to debug?

Reply:
Good point.
Done 5599e1cc3d4aaac7be33ee52e11d278cb86f6286